vaporetto/src/sentence.rs

```diff
@@ -278,964 +318,900 @@ impl Sentence {
+    ///     CharacterBoundary::NotWordBoundary,
+    ///     CharacterBoundary::Unknown,
     /// ], s.boundaries());
     /// ```
-    pub fn boundaries(&self) -> &[BoundaryType] {
+    #[inline]
+    pub fn boundaries(&self) -> &[CharacterBoundary] {
         &self.boundaries
     }
 
-    /// Gets a mutable reference to the boundary information.
-    ///
-    /// # Returns
-    ///
-    /// A mutable reference to the boundary information.
-    pub fn boundaries_mut(&mut self) -> &mut [BoundaryType] {
-        &mut self.boundaries
-    }
-
-    /// Gets a reference to the part-of-speech information.
-    ///
-    /// Each tag is placed at the last of the corresponding token. For example, when the first token
-    /// containing three characters has a tag, that tag will be placed at the third element of the
-    /// returned slice.
-    ///
-    /// # Returns
-    ///
-    /// A reference to the POS information.
+    /// Returns a mutable slice of boundary types.
     ///
     /// # Examples
     ///
     /// ```
-    /// use std::sync::Arc;
+    /// use vaporetto::{CharacterBoundary, Sentence};
     ///
-    /// use vaporetto::{BoundaryType, Sentence};
-    ///
-    /// let s = Sentence::from_tokenized("I/PRP am a/DT cat/NN ./.").unwrap();
-    /// assert_eq!(&[
-    ///     Some(Arc::new("PRP".to_string())), // 'I'
-    ///     None,                             // 'a'
-    ///     None,                             // 'm'
-    ///     Some(Arc::new("DT".to_string())),  // 'a'
-    ///     None,                             // 'c'
-    ///     None,                             // 'a'
-    ///     Some(Arc::new("NN".to_string())),  // 't'
-    ///     Some(Arc::new(".".to_string())),   // '.'
-    /// ], s.tags());
+    /// let mut s = Sentence::from_partial_annotation("火-星|に|行-き|ま-し た").unwrap();
+    /// s.boundaries_mut()[6] = CharacterBoundary::WordBoundary;
+    /// let mut buf = String::new();
+    /// s.write_partial_annotation_text(&mut buf);
+    /// assert_eq!("火-星|に|行-き|ま-し|た", buf);
     /// ```
-    pub fn tags(&self) -> &[Option<Arc<String>>] {
-        &self.tags
+    #[inline]
+    pub fn boundaries_mut(&mut self) -> &mut [CharacterBoundary] {
+        &mut self.boundaries
     }
 
-    /// Gets a mutable reference to the part-of-speech information.
-    ///
-    /// # Returns
-    ///
-    /// A mutable reference to the part-of-speech information.
-    pub fn tags_mut(&mut self) -> &mut [Option<Arc<String>>] {
-        &mut self.tags
+    /// Returns a slice of boundary scores.
+    #[inline]
+    pub fn boundary_scores(&self) -> &[i32] {
+        if self.boundary_scores.is_empty() {
+            &[]
+        } else {
+            &self.boundary_scores[self.score_padding..self.score_padding + self.boundaries.len()]
+        }
     }
 
-    /// Gets a reference to the characters.
-    ///
-    /// # Returns
-    ///
-    /// A reference to the characters.
+    /// Returns a reference to the internal representation of tags.
```

Comment:
```suggestion
    /// Returns a reference to the internal representation of tags.
    ///
    /// In the representation, tags are stored in an array, and
    /// the `j`-th tag of the `i`-th character is stored in the `i*k+j`-th element,
    /// where `k` is the maximum number of tags (i.e., [`n_tags()`]).
```
It would be helpful if there is a description of the internal representation.